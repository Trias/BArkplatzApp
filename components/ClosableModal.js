import React from 'react';
import { ScrollView, Text, Button, View, Modal, TouchableOpacity, StyleSheet } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

export default class ClosableModal extends React.Component {
    render() {
        let {title, onAction, children, actionText, onClose, ...rest} = this.props;
        return <Modal {...rest}>
            <ScrollView contentContainerStyle={styles.scrollViewContainer}>
                <TouchableOpacity style={styles.closeIconContainer} onPress={() => {
                    onClose();
                }}>
                    <MaterialCommunityIcons name="close" style={styles.closeIcon} />
                </TouchableOpacity>
                <Text style={{ fontWeight: 'bold', fontSize: 16, lineHeight: 44 }}>{title}</Text>
                {children}
                <View style={{ marginTop: 20 }}>
                    <Button
                        title={actionText}
                        onPress={() => {
                            onAction && onAction();
                            onClose();
                        }} />
                </View>
            </ScrollView>
        </Modal>
    }
}

const styles = StyleSheet.create({
    closeIconContainer: { position: 'absolute', top: 30, right: 0, lineHeight: 40, zIndex: 100 },
    closeIcon: { fontSize: 30, height: 40, width: 40 },
    scrollViewContainer: { paddingTop: 22, paddingBottom: 44, alignItems: "center", width: '100%', position: 'relative' }
});