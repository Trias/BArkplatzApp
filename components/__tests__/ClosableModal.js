import React from 'react';
import renderer from 'react-test-renderer';

import ClosableModal from '../ClosableModal';

it(`renders correctly`, () => {
  const tree = renderer.create(<ClosableModal 
    actionText={'action'}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
