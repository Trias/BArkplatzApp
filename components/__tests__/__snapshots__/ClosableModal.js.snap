// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  hardwareAccelerated={false}
  visible={true}
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "paddingBottom": 44,
        "paddingTop": 22,
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "lineHeight": 40,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 30,
            "zIndex": 100,
          }
        }
      >
        <Text />
      </View>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "lineHeight": 44,
          }
        }
      />
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              action
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</Modal>
`;
